### DISCUSSION TITLE
Using coverlet without having to modify project files in solution

### DISCUSSION THREAD

**Discussion by awa5114** [coverlet-coverage/coverlet#1697] [Q&A] [answered]
Currently, this is my workflow for generating test coverage reports for a third party project that uses [central package management](https://learn.microsoft.com/en-us/nuget/consume-packages/central-package-management) and a `Directory.Packages.Props` file:

 - `cd` to the test project for which I want to generate a report
 - Add `coverlet.collector` package to test project, and then remove the version
      
 ```
 dotnet add package coverlet.collector
  ```
 - Open the `Directory.Packages.Props` file in a text editor and add the following item group:

  ```xml
  <ItemGroup>
     <PackageVersion Include="coverlet.collector" Version="6.0.0" />
  </ItemGroup>
  ```

 - Run the following command from the test project directory:

  ```
  dotnet test --collect:"XPlat Code Coverage"
  ```

  - Generate a report as follows:

  ```
  reportgenerator.exe -reports:".\TestResults\<guid>\coverage.cobertura.xml"  -targetdir:"coveragereport" -reporttypes:Html
  ```

The issue here is that I basically have to go in and edit the files for the third party project every time I want to generate a report. I do my work with the help of the coverage report but then I have to delete the edits I make to the various files (test project files and the global `Directory.Packages.Props`.

This is quite clunky and an obstacle to generating quick coverage runs to just see what kind of test coverage I have for a specific class or subset of tests.

Is there a way to use coverlet *without* having to modify any project files?

**Comment by Bertk** [comment, marked as answer]
Please use **coverlet.console** instead of **coverlet.collector**. 

https://github.com/coverlet-coverage/coverlet/blob/master/Documentation/GlobalTool.md

**Comment by Bertk** [comment]
Coverlet project uses Directory.Packages.props.

I have also created a simple sample project : https://github.com/Bertk/coverlet-examples/blob/main/Directory.Packages.props
